website/content/docs/internals/integrated-storage.mdx

```diff
@@ -186,3 +242,29 @@ failure scenario.
     </tr>
   </tbody>
 </table>
+
+### Minimums & Scaling
+
+The [Vault Reference Architecture](https://learn.hashicorp.com/tutorials/vault/raft-reference-architecture#recommended-architecture) 
+recommends a 5 node cluster to ensure a minimum failure tolerance of at least 2.
+
+It is good practise, wherever possible, to retain a failure tolerance of 2 or 
+more.
+
+A scaling approach can be pursued in the event of maintenance and other changes 
```

Comment:
I think it's worth noting that autopilot auto-upgrades make this type of scenario much easier. If autopilot auto-upgrades are in use (which is the default in any enterprise cluster 1.11.0+), customers can use their own semantic version compatible version string for the `autopilot_upgrade_version` configuration key and roll clusters at will, just by starting new nodes and waiting for autopilot to do the rest. [These docs](https://www.vaultproject.io/docs/enterprise/automated-upgrades), particularly the section on blue/green style deploys, have more detail.

Reply:
I dont think any further additions are needed at the end - rather we are just stating to maintain a size were failure tolerance are respected and autopilot can have no bearing on the user choosing for example to scale below the advise reference architectures.

What's more irrespective of the enhancements and features for blue/green style deployments - user can choose to deploy in sets which break quorum onto even sets - as for example with 3 to 4 node transitions or anything else which is either below min failure tolerance of 2 or redundantly even.

PS - I've seen other docs out there signifying odd parity and appropriate scaling too and I think its not bad to mention this +2 that's not so obvious for everyone.